Calculator/Calculator.cpp

```diff
@@ -82,97 +84,147 @@ Token Token_stream::get()
     }
 }
 
 //------------------------------------------------------------------------------
+// Gets next token from token stream and calculates a factorial denoted by '!' and
+// returns the value for the factorial. All tokens other than '!' are putback into the
+// token stream
+
+double factorial() {
+    double left = primary();
+    Token t = ts.get();
+    while (true) {
+
+        switch (t.kind) {
+        case '!':
+        {
+            int x = left;
+            int k = (x - 1);
+            double f = x;
+            for (int i = 0; i < k; ++i) {
+                f *= (--x);
+            }
+            return f;
+        }
+        default:
+            ts.putback(t);
+            return left;
+        }
+    }
+}
+
+//------------------------------------------------------------------------------
+// Gets next token from token stream and evaluates a term. Calculates multiplication or division
+// of a term by a factorial or primary and returns the result. All tokens other than '*' and '/'
+// are put back into the token stream
 
 double term()
 {
-    double left = primary();
-    Token t = ts.get(); // get the next Token from the Token stream
+    double left = factorial();
+    Token t = ts.get();
     while (true) {
         switch (t.kind) {
         case '*':
-            left *= primary();
+        {
+            left *= factorial();
             t = ts.get();
             break;
+        }
         case '/':
-        { double d = primary();
+        { double d = factorial();
         if (d == 0) error("divide by zero");
         left /= d;
         t = ts.get();
         break;
         }
         default:
-            ts.putback(t); // put t back into the Token stream
+            ts.putback(t); 
             return left;
         }
     }
 }
 
 
 //------------------------------------------------------------------------------
+// Gets next token from the token stream and evaluates expressions. Calculates 
+// addition and subtraction of terms within the expression. All tokens other than
+// '+' and '-' are returned to the token stream
 
 double expression()
 {
-    double left = term(); // read and evaluate a Term
-    Token t = ts.get(); // get the next Token from the Token stream
+    double left = term(); 
+    Token t = ts.get(); 
     while (true) {
         switch (t.kind) {
         case '+':
-            left += term(); // evaluate Term and add
+            left += term(); 
             t = ts.get();
             break;
         case '-':
-            left -= term(); // evaluate Term and subtract
+            left -= term(); 
             t = ts.get();
             break;
         default:
-            ts.putback(t); // put t back into the token stream
-            return left; // finally: no more + or �; return the answer
+            ts.putback(t); 
+            return left; 
         }
     }
 }
 
 //------------------------------------------------------------------------------
 
+
 int main()
```

Comment:
no overview for main? poor main... lol don't worry about it for now

Reply:
yeah. i was worried about over commenting for some reason so i tried to stick to mainly functions that were doing something, but i can definitely add comment for main as we go along lol